src/server/systems/NetworkingServer.cpp

```diff
@@ -168,23 +174,41 @@ void NetworkingServer::handleUnreliableCommand(sf::Socket::Status status, sf::Pa
             break;
         }
         default: {
-            std::cout << "Unknown command code sent: " << unreliable_command.command << std::endl;
+            std::cerr << "Unknown command code sent: " << unreliable_command.command << std::endl;
             break;
         }
     }
 }
 
 void NetworkingServer::setClientUnreliableUpdate(sf::Packet packet, int client_id,
                                                  uint client_tick) {
+    sf::Uint32 player_id;
+    {
+        std::lock_guard<std::mutex> m_clientToPlayerIds_guard(m_clientToPlayerIds_lock);
+        if (m_clientToPlayerIds_t.count(client_id) == 0) {
+            return;
+        }
+        player_id = m_clientToPlayerIds_t[client_id];
+    }
+
     int drift = std::abs(static_cast<int>((m_tick_ta - client_tick)));
+    {
+        std::lock_guard<std::mutex> m_playerIdsToDriftMetric_guard(m_playerIdsToDriftMetrics_lock);
+        if (m_playerIdsToDriftMetrics_t.find(player_id) == m_playerIdsToDriftMetrics_t.end()) {
+            m_playerIdsToDriftMetrics_t[player_id] = new TemporalMetric{120, sf::seconds(0.5f)};
+        }
+        m_playerIdsToDriftMetrics_t[player_id]->pushCount(drift);
+        m_playerIdsToDriftMetrics_t[player_id]->update();
+
+    }
     if (drift < CMD_DRIFT_THRESHOLD) {
-        sf::Uint32 player_id;
         {
-            std::lock_guard<std::mutex> m_clientToPlayerIds_guard(m_clientToPlayerIds_lock);
-            if (m_clientToPlayerIds_t.count(client_id) == 0) {
-                return;
+            std::lock_guard<std::mutex> m_playerIdsToUpdatesMetric_guard(m_playerIdsToUpdatesMetric_lock);
```

Comment:
code is duplicated but only once and it's clear what it does vs deduping might make it less clear, leaning towards leaving as is until we have more code like this